ci/bin/format.dart

```diff
@@ -0,0 +1,795 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Checks and fixes format on files with changes.
+//
+// Usage: dart format.dart [--fix] [--all]
+//   --fix will fix all of the formatting problems found.
+//   --all will find/fix formatting problems for all files, not just changed
+//         files.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+import 'package:process_runner/process_runner.dart';
+
+class FormattingException implements Exception {
+  FormattingException(this.message, [this.result]);
+
+  final String message;
+  final ProcessResult /*?*/ result;
```

Comment:
Aren't we on an NNBD enabled SDK already?

Reply:
Yes, but when I tried to turn that on, it wanted all of the dependent packages to be NNBD too. Maybe I was holding it wrong, though.

Reply:
Ahh. Maybe we can just migrate all the stuff  in here at once then.  That's fine.